cheddar/cheddar-integration-aws/src/main/java/com/clicktravel/infrastructure/persistence/aws/cloudsearch/AmazonCloudSearchDomainClientBuilder.java

```diff
@@ -16,15 +16,13 @@
  */
 package com.clicktravel.infrastructure.persistence.aws.cloudsearch;
 
-import com.amazonaws.auth.AWSCredentials;
 import com.amazonaws.services.cloudsearchdomain.AmazonCloudSearchDomain;
 import com.amazonaws.services.cloudsearchdomain.AmazonCloudSearchDomainClient;
 
 public class AmazonCloudSearchDomainClientBuilder {
 
-    public static AmazonCloudSearchDomain build(final AWSCredentials awsCredentials, final String endpoint) {
-        final AmazonCloudSearchDomain amazonCloudSearchDomain = awsCredentials == null
-                ? new AmazonCloudSearchDomainClient() : new AmazonCloudSearchDomainClient(awsCredentials);
+    public static AmazonCloudSearchDomain build(final String endpoint) {
```

Comment:
Removing this class makes unit testing of `CloudSearchEngine` very difficult, as two clients are (identically) constructed and used for different purposes. Other simplifications of this class will make this more tractable, so I would prefer to defer this for now.